File: src/main/java/io/github/saneea/dvh/feature/process/Proc.java
package io.github.saneea.dvh.feature.process;

import java.util.Map;
import java.util.function.Supplier;

import io.github.saneea.dvh.Feature;
import io.github.saneea.dvh.FeatureContext;
import io.github.saneea.dvh.feature.multi.MultiFeatureBase;

public class Proc extends MultiFeatureBase {

	@Override
	public Meta meta(FeatureContext context) {
		return Meta.from("fork process");
	}

	@Override
	public Map<String, Supplier<Feature>> getFeatureAliases() {
		return new AliasesBuilder()//
				.feature("inspect", InspectProcess::new)//
				.feature("toFile", ToFile::new)//
				.build();
	}
}

File: src/main/java/io/github/saneea/dvh/feature/process/Proc.kt
package io.github.saneea.dvh.feature.process

import io.github.saneea.dvh.Feature.Meta
import io.github.saneea.dvh.FeatureContext
import io.github.saneea.dvh.feature.multi.FeaturesCreators
import io.github.saneea.dvh.feature.multi.MultiFeatureBase

class Proc : MultiFeatureBase() {
    override fun meta(context: FeatureContext) = Meta("fork process")

    override fun getFeatureAliases(): FeaturesCreators =
        AliasesBuilder()
            .feature("inspect", ::InspectProcess)
            .feature("toFile", ::ToFile)
            .build()
}